quran/api/views.py

```diff
@@ -0,0 +1,17 @@
+from rest_framework import viewsets
+
+from ..models import Surah
+from .serializers import SurahListSerializer, SurahDetailsSerializer
+
+
+class SurahListRetrieveView(viewsets.ReadOnlyModelViewSet):
+    """View that allows to List, Retrieve Surah."""
+    lookup_field = 'number'
+    queryset = Surah.objects.all()
+    serializer_class = SurahListSerializer
+
+    def get_serializer_class(self):
+        """Returns appreciate serializer class for the given action."""
+        if self.action == "retrieve":
+            return SurahDetailsSerializer
+        return super().get_serializer_class()
```

Comment:
Actually this returns `serializer_class` which we set on view (SurahListSerializer). Yeap we also can write

`return SurahListSerializer` instead of   `return super().get_serializer_class()`. But by this way we show that it's overwritten function.